src/nbl/CMakeLists.txt

```diff
@@ -469,6 +469,18 @@ target_include_directories(Nabla PUBLIC
 	"${THIRD_PARTY_SOURCE_DIR}/dxc/dxc/external/SPIRV-Tools/include"
 	"${THIRD_PARTY_SOURCE_DIR}/dxc/dxc/external/SPIRV-Headers/include"
 )
+
+# nlohmann_json
+set(JSON_BuildTests OFF CACHE INTERNAL "")
+add_dependencies(Nabla nlohmann_json)
+if(NBL_STATIC_BUILD)
+	target_link_libraries(Nabla INTERFACE nlohmann_json::nlohmann_json)
+else()
+	target_link_libraries(Nabla PRIVATE nlohmann_json::nlohmann_json)
+endif()
+target_include_directories(Nabla PUBLIC ${THIRD_PARTY_SOURCE_DIR}/nlohmann_json/include)
```

Comment:
make it

```cmake
target_include_directories(Nabla PUBLIC $<TARGET_PROPERTY:nlohmann_json::nlohmann_json,INCLUDE_DIRECTORIES>)
```
 this way if a target uses a set of include directories which are auto-generated or depend on the configuration you will catch them all properly

Reply:
I'm not really understanding this setting. If I do this it won't find the headers

Reply:
also change scope from `PUBLIC` to `PRIVATE` in `target_include_directories`

Reply:
> I'm not really understanding this setting

this is a [generator expression](https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#manual:cmake-generator-expressions(7)) which gets [`INCLUDE_DIRECTORIES`](https://cmake.org/cmake/help/latest/prop_tgt/INCLUDE_DIRECTORIES.html#prop_tgt:INCLUDE_DIRECTORIES) target property storing `nlohmann_json::nlohmann_json`'s target set of include search directories, the expression is evaluated during build system generation & can handle configuration specific defined paths (it may have paths which are different depending on a configuration). Doing so is handy because it's a target responsibility to define the includes not ours

Reply:
@AnastaZIuk review if fixed

Reply:
will correct myself later, resolving